woocommerce/woocommerce-android
# `ProductSelector` - filter out redundant products

## Filtering out redundant product items from `ProductSelector` list.

Closes: #8503 


### Description

This PR removes products that are not published and variable products with no variations from the `ProductSelector`'s list.

### Testing instructions

1. Go to Orders and create new order
2. Click "+ Add product"
3. Verify that only valid products are displayed on the list. There should not be any draft products (unpublished state) or variable products with no variations.

We'll update it in the final PR to trunk.
